### PR TITLE
More explicit design pattern documentation for multi-state modals

### PR DESCRIPTION
**Pull request by ErnieAtLYD**
After some back and forth with folks in RED, other developer designers and members of the billing team, there was a consensus that more explicit documentation was needed covering design patters in multi-state modal windows. This attempts to address this in http://rackerlabs.github.io/encore-ui/#/components/rxModalAction.

### LGTMs
- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM

### CODE REVIEW

File: src/components/rxModalAction/docs/rxModalAction.html [hunk 1]
```diff
@@ -1,5 +1,9 @@
 <!-- Sample HTML goes here as a live example of how to the component can be used -->
 <div ng-controller="rxModalActionCtrl">
+
+    <h3>Regular Modals</h3>
+    <p>Use modals whenever a user needs to take an action and you their full attention to do so before interacting with the original page, or you want to interrupts a user’s current task to catch the user’s full attention to something more important.</p>
```
Comment (CITguy) [inline src/components/rxModalAction/docs/rxModalAction.html]: * Per the [HTML Styleguide](https://github.com/rackerlabs/encore-ui/blob/master/guides/html-styleguide.md#html-max-line-length), Lines must be a max of 120 characters (80 preferred)

File: src/components/rxModalAction/docs/rxModalAction.html [hunk 2]
```diff
@@ -13,6 +17,10 @@
         </rx-modal-action>
     </p>
 
+    <h3>Multi-View Modals</h3>
+    <p>A multi-state modal breaks apart complex interactions so modal content isn’t too long. Multi-state modals also work well when the user must complete multiple sequential steps.</p>
```
Comment (CITguy) [inline src/components/rxModalAction/docs/rxModalAction.html]: * Per the [HTML Styleguide](https://github.com/rackerlabs/encore-ui/blob/master/guides/html-styleguide.md#html-max-line-length), Lines must be a max of 120 characters (80 preferred)
* You seem to have accidentally pasted "smart" quotes in your content. Please correct (apostrophe in `isn't`).

File: src/components/rxModalAction/docs/rxModalAction.html [hunk 3]
```diff
@@ -13,6 +17,10 @@
...
+    <p>It works for cases when removing or deleting something needing a confirmation click, such as "Are You Sure you want to Remove this?”</p>
```
Comment (CITguy) [inline src/components/rxModalAction/docs/rxModalAction.html]: * Per the [HTML Styleguide](https://github.com/rackerlabs/encore-ui/blob/master/guides/html-styleguide.md#html-max-line-length), Lines must be a max of 120 characters (80 preferred)
* You seem to have accidentally pasted "smart" quotes in your content. Please correct (opening/closing double quotes).

File: src/components/rxModalAction/docs/rxModalAction.html [hunk 4]
```diff
@@ -113,4 +121,13 @@
         </rx-modal-footer>
     </script>
 
+    <h3>Do's and Don'ts</h3>
+    <ul>
+        <li>Do use modals to perform actions requiring the users full attention.</li>
+        <li>Don't display a modal without an action button, or a way to close the modal by closing the modal via X or pressing the Cancel button.</li>
+        <li>Don't use modals as pop-ups, having them appear without the user clicking something</li>
+        <li>Don’t use modals referring to information on the originating page</li>
```
Comment (CITguy) [inline src/components/rxModalAction/docs/rxModalAction.html]: * You seem to have accidentally pasted "smart" quotes in your content. Please correct (apostrophe in `Don't`)

**Comment by ErnieAtLYD** [comment]
Sorry about that. Typos fixed.

**Comment by CITguy** [comment]
LGTM

@glynnis, do you have any comments?

File: src/components/rxModalAction/docs/rxModalAction.html [hunk 5]
```diff
@@ -1,5 +1,12 @@
 <!-- Sample HTML goes here as a live example of how to the component can be used -->
 <div ng-controller="rxModalActionCtrl">
+
+    <h3>Regular Modals</h3>
+    <p>Use modals whenever a user needs to take an action and you their full
```
Comment (glynnis) [inline src/components/rxModalAction/docs/rxModalAction.html]: `and you their full` -> `that needs their`

File: src/components/rxModalAction/docs/rxModalAction.html [hunk 6]
```diff
@@ -1,5 +1,12 @@
...
+    attention to do so before interacting with the original page, or you want
```
Comment (glynnis) [inline src/components/rxModalAction/docs/rxModalAction.html]: > to do so before interacting with the original page

I don't understand what this means. I think you can just cut this.

File: src/components/rxModalAction/docs/rxModalAction.html [hunk 7]
```diff
@@ -1,5 +1,12 @@
...
+    to interrupts a user's current task to catch the user's full attention to
```
Comment (glynnis) [inline src/components/rxModalAction/docs/rxModalAction.html]: `interrupts` -> `interrupt`

`to catch the user's full attention` -> `to redirect their attention`

File: src/components/rxModalAction/docs/rxModalAction.html [hunk 8]
```diff
@@ -13,6 +20,14 @@
         </rx-modal-action>
     </p>
 
+    <h3>Multi-View Modals</h3>
+    <p>A multi-state modal breaks apart complex interactions so modal content
+    isn't too long. Multi-state modals also work well when the user must
```
Comment (glynnis) [inline src/components/rxModalAction/docs/rxModalAction.html]: Can you change all instances of `multi-state` to `multi-view` just so we have consistency with the title/don't introduce duplicate terms?

**Comment by ErnieAtLYD** [comment]
Thanks for keeping me honest, @glynnis - changes made.

**Comment by glynnis** [comment]
Design LGTM!

🎉 👏 🍰 💯 🏆  Thanks for submitting this @ErnieAtLYD. You da best.

**Comment by parlarjb** [comment]
LGTM

**Pull request merged**